android/app/src/main/java/org/jitsi/meet/AudioModeModule.java

```diff
@@ -0,0 +1,440 @@
+        if (enabled) {
+            audioManager.startBluetoothSco();
+            audioManager.setBluetoothScoOn(true);
+        } else {
+            audioManager.setBluetoothScoOn(false);
+            audioManager.stopBluetoothSco();
+        }
+    }
+
+    /**
+     * Helper method to trigger an audio route update when devices change.  It makes sure the
+     * operation is performed on the main thread.
+     */
+    private void onAudioDeviceChange() {
+        mainThreadHandler.post(mainThreadRunner);
+    }
+
+    /**
+     * Helper class to detect and handle Bluetooth device changes.  It monitors Bluetooth headsets
+     * being connected / disconnected and notifies the module about device changes when this occurs.
+     */
+    private class BluetoothManager {
+        /**
+         * Reference to the Bluetooth adapter, needed for managing <tt>BluetoothProfile.HEADSET</tt>
+         * devices.
+         */
+        private BluetoothAdapter bluetoothAdapter;
+
+        /**
+         * Reference to a proxy object which allows us to query connected devices.
+         */
+        private BluetoothHeadset bluetoothHeadset;
+
+        /**
+         * Listener for Bluetooth service profiles, allows us to get the proxy object to
+         * {@link BluetoothHeadset}.
+         */
+        private BluetoothProfile.ServiceListener bluetoothProfileListener;
+
+        /**
+         * Helper for running Bluetooth operations on the main thread.
+         */
+        private Runnable bluetoothRunnable;
+
+        /**
+         * Flag indicating if there are any Bluetooth headset devices currently available.
+         */
+        private boolean headsetAvailable = false;
+
+        /**
+         * Start monitoring Bluetooth device activity.
+         */
+        public void start() {
+            ReactContext reactContext = getReactApplicationContext();
+            if (reactContext == null) {
+                return;
+            }
+
+            bluetoothRunnable = new Runnable() {
+                @Override
+                public void run() {
+                    if (bluetoothHeadset == null) {
+                        headsetAvailable = false;
+                    } else {
+                        List<BluetoothDevice> devices = bluetoothHeadset.getConnectedDevices();
+                        headsetAvailable = !devices.isEmpty();
+                    }
+                    onAudioDeviceChange();
+                }
+            };
+
+            bluetoothAdapter = BluetoothAdapter.getDefaultAdapter();
+            if (bluetoothAdapter == null) {
+                Log.w(TAG, "Device doesn't support Bluetooth");
+                return;
+            }
+
+            if (!audioManager.isBluetoothScoAvailableOffCall()) {
+                Log.w(TAG, "Bluetooth SCO is not available");
+                return;
+            }
+
+            bluetoothProfileListener = new BluetoothProfile.ServiceListener() {
```

Comment:
Both `bluetoothProfileListener` and `bluetoothReceiver` appear to be dealing with connected and disconnected  and calling `updateDevices` but I have no idea why any of these 2 are necessary, why 1 of them alone is insufficient. What I probably need is a sentence of comment stating what these are doing.